src/Mover.ts

```diff
@@ -796,24 +834,103 @@ export class MoverAPI implements Types.MoverAPI {
 
-        if (!focused || (await this._isIgnoredInput(focused, keyCode))) {
+        if (focused && (await this._isIgnoredInput(focused, keyCode))) {
             return;
         }
 
-        const ctx = RootAPI.getTabsterContext(tabster, focused, {
-            checkRtl: true,
-        });
+        const ctx =
+            focused &&
+            RootAPI.getTabsterContext(tabster, focused, {
+                checkRtl: true,
+            });
 
-        if (
-            !ctx ||
-            !ctx.mover ||
-            ctx.isExcludedFromMover ||
-            ctx.ignoreKeydown(event)
-        ) {
+        const mover = ctx?.mover;
+
+        if (!mover) {
+            // Mover's nextFor property allows the Mover to gain focus when
+            // the arrow keys are pressed on an element that matches the
+            // nextFor selector. Here we find if there is a Mover that wants
+            // to gain focus given the currently focused element.
+            const moversFor = this._moversFor;
+            const allMoversFor: [HTMLElement | undefined, Mover?][] = [];
+            const focusedOrBody = focused || win.document.body;
+
+            for (const nextFor of Object.keys(moversFor)) {
+                if (matchesSelector(focusedOrBody, nextFor)) {
+                    for (const m of moversFor[nextFor]) {
+                        allMoversFor.push([m.getElement(), m]);
+                    }
+                }
+            }
+
+            if (allMoversFor.length > 0) {
+                allMoversFor.push([focusedOrBody]);
+
+                // Sort by DOM position to find the closest Mover when there are
+                // multiple matching Movers.
+                allMoversFor.sort((a, b) => {
+                    const aElement = a[0];
+                    const bElement = b[0];
+
+                    return aElement &&
+                        bElement &&
+                        aElement.compareDocumentPosition(bElement) &
+                            Node.DOCUMENT_POSITION_FOLLOWING
+                        ? -1
+                        : 1;
+                });
+
+                const isBackward =
+                    keyCode === Keys.Up ||
+                    keyCode === Keys.Left ||
+                    keyCode === Keys.PageUp ||
+                    keyCode === Keys.Home;
+
+                const focusedIndex = allMoversFor.findIndex(
+                    (a) => a[0] === focusedOrBody
+                );
+
+                let moverToFocus: Mover | undefined;
+
+                if (focusedIndex === 0) {
+                    moverToFocus = allMoversFor[1][1];
+                } else if (focusedIndex === allMoversFor.length - 1) {
+                    moverToFocus = allMoversFor[focusedIndex - 1][1];
+                } else {
+                    moverToFocus =
+                        allMoversFor[
+                            isBackward ? focusedIndex - 1 : focusedIndex + 1
+                        ][1];
+                }
```

Comment:
Perhaps this can be broken into a `initializeMoverTofocus` helper method?! To help reducing nesting